docs/guide/snaps-introduction.md

```diff
@@ -0,0 +1,69 @@
+# Snaps Tutorial
+
+## What is Snaps?
+
+Snaps are a new evolution of MetaMask's functionality. By using MetaMask Flask in conjunction with the Snaps API, developers are able to bring their own code to the wallet in a custom-made code laboratory. MetaMask is the first wallet to provide this openness and flexibility to developers, paving the way to scale the number of features and personalization available to the wallet and your Dapp.
+
+Snaps allows developers to build and plug in their own APIs to MetaMask. No longer do you have to rely on MetaMask to roll out features and updates. The rate of innovation, and MetaMask's flexibility, are limited only by your imagination and your capacity to build. 
+
+## MetaMask Snaps: Overview
+
+Broadly speaking, MetaMask Snaps consists of two things: (1) a way to run untrusted JavaScript inside the MetaMask application, and (2) APIs for websites and MetaMask to communicate with individual snaps. As with MetaMask’s [Ethereum JavaScript provider API](https://docs.metamask.io/guide/ethereum-provider.html), communication occurs via JSON-RPC requests and responses.
```

Comment:
I find the division of snaps into "two things" here as a very odd way to frame things.

I would describe snaps as "a way to run untrusted JavaScript inside the MetaMask application", that part is good. Then maybe we could go on to explain the API that we offer to snaps, and that we offer APis that allow websites to communicate with snaps. I can give this a shot later.

Reply:
On further consideration I think it was just the specific wording here that was throwing me off. Generally this isn't bad.

Here's what I came up with trying to re-write it. Not sure that it's better though.

>Broadly speaking, MetaMask Snaps is a system that allows running untrusted programs inside MetaMask and granting them specific capabilities. Snaps are run in an isolated environment where they have access to a limited set of capabilities, determined by the permissions they were granted by the user during installation. As with MetaMask’s [Ethereum JavaScript provider API](https://docs.metamask.io/guide/ethereum-provider.html), snaps communicate with MetaMask using a JSON-RPC API.
>
>New RPC methods have been added to our [Ethereum JavaScript provider API](https://docs.metamask.io/guide/ethereum-provider.html) as well. These methods allow websites to request that snaps be installed, and to communicate with individual snaps.

Reply:
> On further consideration I think it was just the specific wording here that was throwing me off. 

I have this experience a lot when collaborating on a text, so I feel you

Your version gives a lot more detail, and given that we're writing for devs here to explain how to do something complex, that seems appropriate to me.